sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs

```diff
@@ -405,6 +405,98 @@ public virtual BlobClient GetBlobClient(string blobName)
                 EncryptionScope);
         }
 
+        /// <summary>
+        /// Create a new <see cref="BlockBlobClient"/> object by
+        /// concatenating <paramref name="blobName"/> to
+        /// the end of the <see cref="Uri"/>. The new
+        /// <see cref="BlockBlobClient"/>
+        /// uses the same request policy pipeline as the
+        /// <see cref="BlobContainerClient"/>.
+        /// </summary>
+        /// <param name="blobName">The name of the block blob.</param>
+        /// <returns>A new <see cref="BlockBlobClient"/> instance.</returns>
+        protected internal virtual BlockBlobClient GetBlockBlobClientInternal(string blobName)
```

Comment:
We should use a different suffix like `Core` instead of `Internal`.  I think we have a similar `WithSnapshotCore`?